ftp-sink/src/test/java/org/springframework/cloud/stream/module/ftp/FtpSinkIntegrationTests.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2015 the original author or authors.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.module.ftp;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.util.Properties;
+
+import org.hamcrest.Matchers;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.cloud.stream.messaging.Sink;
+import org.springframework.cloud.stream.modules.test.PropertiesInitializer;
+import org.springframework.cloud.stream.modules.test.file.remote.TestFtpServer;
+import org.springframework.messaging.support.GenericMessage;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+
+/**
+ * @author David Turanski
+ * @author Marius Bogoevici
+ * @author Gary Russell
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = FtpSinkApplication.class, initializers = PropertiesInitializer.class)
+@DirtiesContext
+public class FtpSinkIntegrationTests extends TestFtpServer {
+
+	@BeforeClass
+	public static void configureFtpServer() throws Throwable {
+
+		Properties properties = new Properties();
+		properties.put("remoteDir", "ftpTarget");
+		properties.put("username", "foo");
+		properties.put("password", "foo");
+		properties.put("filenamePattern", "*");
+		properties.put("port", port);
+		properties.put("mode", "FAIL");
+		properties.put("filenameExpression", "payload.name.toUpperCase()");
+		PropertiesInitializer.PROPERTIES = properties;
```

Comment:
I wonder if we can rely on the `System.properties` or even `ENV` variables...
Do you remember the [12 Factor App](http://12factor.net/) principles ?